tests/test_quantity.py

```diff
@@ -143,14 +143,20 @@ def test_to_19():
 
 def test_to_20():
     v = q.Quantity(1.0, "Hz")
-    with pytest.raises(ValueError) as e_info:
-        convert = v.to("radian s^-1")
+    try:
+        v.to("radian s^-1")
+        raise q.QuantityError("Hz", "radian s^-1")
+    except q.QuantityError as e:
+        assert e.value_error == q.QuantityError("Hz", "radian s^-1").value_error
 
 
 def test_to_21():
     v = q.Quantity(1.0, "radian s^-1")
-    with pytest.raises(ValueError) as e_info:
-        convert = v.to("Hz")
+    try:
+        v.to("Hz")
+        raise q.QuantityError("radian s^-1", "Hz")
```

Comment:
@hpohekar rather than comparing the exceptions directly, you can compare the exception data. Comparing exceptions is not reliable. That's the only thing you need to change - up to now you were comparing the message strings.

Reply:
```

>>> class UnitConversionError(ValueError):
...   def __init__(self, from_unit, to_unit):
...     self.from_unit = from_unit
...     self.to_unit = to_unit
       # ...
...

>>> with pytest.raises(UnitConversionError) as ee:
...   raise UnitConversionError("mm", "lb")
...
>>> ee.value.to_unit
'lb'
>>> ee.value.from_unit
'mm'
```


```

Reply:
Done. Thank you.